tracer/src/Datadog.Trace/Debugger/DynamicInstrumentation.cs

```diff
@@ -87,87 +82,76 @@ internal DynamicInstrumentation(
 
         public bool IsDisposed => Volatile.Read(ref _disposeState) != 0;
 
-        public bool IsInitialized => Volatile.Read(ref _initState) == 2;
+        public bool IsInitialized { get; private set; }
 
         internal void Initialize()
         {
-            var originalState = Interlocked.CompareExchange(ref _initState, 1, 0);
-
-            // If we weren't in "not initialized" state, return early
-            if (originalState != 0)
+            if (!_settings.DynamicInstrumentationEnabled)
             {
+                Log.Information("Dynamic Instrumentation is disabled. To enable it, please set DD_DYNAMIC_INSTRUMENTATION_ENABLED environment variable to 'true'.");
                 return;
             }
 
-            try
-            {
-                if (!_settings.DynamicInstrumentationEnabled)
-                {
-                    Log.Information("Dynamic Instrumentation is disabled. To enable it, please set DD_DYNAMIC_INSTRUMENTATION_ENABLED environment variable to 'true'.");
-                    // Reset to "not initialized"
-                    Interlocked.Exchange(ref _initState, 0);
-                    return;
-                }
-
-                Log.Information("Dynamic Instrumentation initialization started");
+            Log.Information("Dynamic Instrumentation initialization started");
 
-                // Initialize without blocking
-                _ = Task.Run(async () => await InitializeAsync().ConfigureAwait(false), _cancellationTokenSource.Token);
-            }
-            catch (Exception e)
-            {
-                Log.Error(e, "Initializing Dynamic Instrumentation failed.");
-                // Reset to "not initialized"
-                Interlocked.Exchange(ref _initState, 0);
-            }
+            // Start initialization in background
+            _ = Task.Run(InitializeAsync)
```

Comment:
There's so much "in the background" stuff that I get lost 😢 Given _everything_ is started "in the background" anyway in `Instrumentation`, do we _really_ need to make everything explicitly use `Task.Run`? It would be much easier to follow if the `DebuggerManager` for example, could just do something like this:

```csharp
Task allTasks = Task.WhenAll(
                  DynamicInstrumentation.InitializeAsync(),
                  SymbolUploader.InitializeAsync(),
                  SomethingElse.InitializeAsync());
```

Not required, and could be another PR if you want, but I think it would make a lot of these things much easier to follow

Reply:
Deleted some Task.Run in d3b96dbf51b46f1ac593b3faee04ef7803804c7e